### PR TITLE
[ISSUE #4414] Add spring sink connector.

### LINKED ISSUE #4414: [Feature] Add Springboot sink connector
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Feature Request

Add Springboot sink connector

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [ ] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

### PR DESCRIPTION
**Pull request by yanrongzhen**
Fixes #4414 .

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications
Add spring sink connector.
Add `@EventMeshListener` annotation, support annotation add on method, deserialization through method imported parameters.

image

### CODE REVIEW

File: eventmesh-openconnect/eventmesh-openconnect-java/src/main/java/org/apache/eventmesh/openconnect/Application.java [hunk 1]
```diff
@@ -43,8 +45,16 @@ public void run(Class<? extends Connector> clazz) throws Exception {
...
+            String spiKey = SPRING;
+            if (isSink(clazz)) {
+                spiKey += Constants.SINK;
+            } else if (isSource(clazz)) {
+                spiKey += Constants.SOURCE;
+            } else {
+                throw new EventMeshException("Connector is invalid.");
+            }
+            ConnectorCreateService createService =
+                EventMeshExtensionFactory.getExtension(ConnectorCreateService.class, spiKey);
```
[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-openconnect/eventmesh-openconnect-java/src/main/java/org/apache/eventmesh/openconnect/Application.java:57]: Generally, there will be no other types of loading, the special logic here is due to the need to obtain the connector through the spring container.

---

一般不会再有其他类型的加载, 这里特殊逻辑是由于需要通过spring容器去获取connector.

[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-openconnect/eventmesh-openconnect-java/src/main/java/org/apache/eventmesh/openconnect/Application.java]: I optimized this logic with extensions in the new round of commits.

---

我在新一轮提交中通过扩展优化了此逻辑. @pandaapo

[the remark this answers is not in the document]
Comment (xwm1992) [reply eventmesh-openconnect/eventmesh-openconnect-java/src/main/java/org/apache/eventmesh/openconnect/Application.java]: I think the most fundamental problem here is that the connector in the application is created through reflection, but the implementation of spring is left to the spring bean for management. The application and spring are not well combined, so the spi method is used for loading.
---
我认为这里最根本的问题在于application中是通过反射的方式创建的connector，但是spring的实现交给了spring bean进行管理，application与spring没有很好的进行结合，才采用的spi的方式进行加载spring的connector实例。

File: eventmesh-connectors/eventmesh-connector-spring/src/main/java/org/apache/eventmesh/connector/spring/sink/EventMeshListener.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.stereotype.Component;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Component
+public @interface EventMeshListener {
```
[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-connectors/eventmesh-connector-spring/src/main/java/org/apache/eventmesh/connector/spring/sink/EventMeshListener.java:32]: Fixed

[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-connectors/eventmesh-connector-spring/src/main/java/org/apache/eventmesh/connector/spring/sink/EventMeshListener.java:32]: The above mentioned issues have been fixed.

File: eventmesh-connectors/eventmesh-connector-spring/src/main/java/org/apache/eventmesh/connector/spring/sink/connector/SpringSinkConnector.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+/**
+ * Spring sink connector
+ */
+@Slf4j
+public class SpringSinkConnector implements Sink {
+
+    private SpringSinkConfig sinkConfig;
+
+    private BlockingQueue<ConnectRecord> queue;
```
[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-connectors/eventmesh-connector-spring/src/main/java/org/apache/eventmesh/connector/spring/sink/connector/SpringSinkConnector.java:41]: Spring Source Connector, and Spring Sink Connector, not only transfer in memory, actually provide the ability to send and receive messages for EventMesh through Spring.

---

Spring Source Connector, 和Spring Sink Connector, 不仅是在内存中传输, 实际上是提供了通过Spring的方式, 对于EventMesh进行消息收发的能力.

[the remark this answers is not in the document]
Comment (yanrongzhen) [reply eventmesh-connectors/eventmesh-connector-spring/src/main/java/org/apache/eventmesh/connector/spring/sink/connector/SpringSinkConnector.java:41]: I think in Spring enterprise applications, the integration of message sending/consumption and spring will be more friendly for users who want to access EventMesh using the Spring environment.

---

我觉得在Spring企业级应用当中, 消息的发送/消费和spring进行集成, 会对于使用Spring环境想接入EventMesh用户更加友好.

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**